src/darwin/Framework/CHIP/MTRDeviceControllerFactory.mm

```diff
@@ -742,6 +755,32 @@ - (MTRDeviceController * _Nullable)createControllerOnNewFabric:(MTRDeviceControl
                                   error:error];
 }
 
+- (void)preWarmCommissioningSession
+{
+    dispatch_async(_chipWorkQueue, ^{
+        CHIP_ERROR err = CHIP_ERROR_INCORRECT_STATE;
+        if (!self->_running) {
+            MTR_LOG_ERROR("Can't pre-warm, Matter controller factory is not running");
```

Comment:
Do we still want to record the metric in this case?

Reply:
We'll get the error code in the metric, so I think I'll leave it in for now.